app/src/main/java/com/sdp/movemeet/backend/BackendActivityManager.java

```diff
@@ -35,11 +35,12 @@ public void uploadActivity(Activity activity, OnSuccessListener<Void> onSuccess,
         if (activity.getBackendRef() == null) {
             docRef = getActivitiesCollectionReference().document();
             activity.setBackendRef(docRef);
-            activity.setDocumentPath(docRef.getId());
+            //activity.setDocumentPath(docRef.getId());
         } else {
             docRef = activity.getBackendRef();
-            activity.setDocumentPath(docRef.getId());
+            //activity.setDocumentPath(docRef.getId());
         }
+        activity.setDocumentPath(docRef.getId());
```

Comment:
... This class should be deleted, but nice catch nonetheless!

Reply:
Ok ok, well in the version of the project I have, UploadActivityActivity still requires this BackendActivityManager. That's the reason why I haven't removed it yet!